jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -1556,19 +1557,14 @@ ecma_string_trim (const ecma_string_t *string_p) /**< pointer to an ecma string
 {
   ecma_string_t *ret_string_p;
 
-  lit_utf8_size_t buffer_size = ecma_string_get_size (string_p);
+  ECMA_STRING_TO_UTF8_STRING (string_p, utf8_str_p);
 
-  if (buffer_size > 0)
+  if (utf8_str_p_size > 0)
```

Comment:
Yes, the problem is that at `ECMA_STRING_TO_UTF8_STRING (string_p, utf8_str_p);` nothing hints at `utf8_str_p_size` being declared. (Note: the macro *does* declare other variables as well -- like `is_ascii` and `must_be_freed` -- but they are never used outside of the macros. By "outside" I literally mean outside of the macro definition. That must have been the mistunderstanding: I guess you meant outside the code bracketed by the expansions of `ECMA_STRING_TO_UTF8_STRING` and `ECMA_FINALIZE_UTF8_STRING`.)